### PR TITLE
User type symbol equality everywhere

### PR DESCRIPTION
**Pull request by Arkatufus**
## Changes

* Remove as many magic string as possible by referencing the actual class/methods/properties from the actual compilation instance.
* Use type equality to make sure that each class/methods/properties actually came from the reference Akka assemblies.

### CODE REVIEW

File: src/Akka.Analyzers.Roslyn311/Akka.Analyzers.Roslyn311.csproj [hunk 1]
```diff
@@ -12,5 +12,8 @@
 
     <ItemGroup>
         <Compile Include="..\Akka.Analyzers\**\*.cs" Exclude="..\Akka.Analyzers\obj\**\*.cs" />
+        <Compile Update="..\Akka.Analyzers\Context\Core\Actor\ActorContextContext.cs">
```
Comment (Aaronontheweb) [inline src/Akka.Analyzers.Roslyn311/Akka.Analyzers.Roslyn311.csproj:15]: Why is this necessary @Arkatufus ?

**Review by Aaronontheweb — COMMENTED** [review]
Just 1 question but otherwise looks good

File: src/Akka.Analyzers.Roslyn311/Akka.Analyzers.Roslyn311.csproj (hunk 1 above)
Comment (Arkatufus) [reply src/Akka.Analyzers.Roslyn311/Akka.Analyzers.Roslyn311.csproj:15]: That's a strange refactor artifact by Rider, let me remove that

**Review by Aaronontheweb — APPROVED** [review]
lgtm
